account-kit/react/src/components/auth/card/error/connection-error.tsx

```diff
@@ -86,3 +103,23 @@ export const ConnectionError = ({
     </div>
   );
 };
+
+const clearWalletConnectStore = () => {
+  // Open Wallet Connect Indexed DB
+  const dbOpenRequest = indexedDB.open("WALLET_CONNECT_V2_INDEXED_DB");
+
+  dbOpenRequest.onsuccess = () => {
+    const db = dbOpenRequest.result;
+
+    const txn = db.transaction(["keyvaluestorage"], "readwrite");
+
+    const store = txn.objectStore("keyvaluestorage");
+
+    // Clear Store
+    store.clear();
+  };
+
+  dbOpenRequest.onerror = () => {
+    console.error("Error opening Wallet Connect DB. Cannot clear store.");
+  };
+};
```

Comment:
Adding this function to clear the IndexedDB store wallet connect  populates on attempted authentication. This is to address the fact that we could get into continuous error loops when an auth attempt fails.

Reply:
oh added a comment on the other PR but we might want to add this to the `disconnect` action from `account-kit/core`

Reply:
actually... looking at the definition for wallet connect providers, https://github.com/WalletConnect/walletconnect-monorepo/blob/v2.0/providers/universal-provider/src/UniversalProvider.ts#L546

the session should not be getting used on retry if we're using the `disconnect` action from core which calls the `wagmi_disconnect` method. That method then calls the connectors disconnect which eventually clears the wallet connect session.

Reply:
From my tests, this doesn't seem to be the case. I'd be sending a video recording to explain the behaviour calling `disconnect` from `@account-kit/core`